src/datasources/analytics.ts

```diff
@@ -0,0 +1,42 @@
+import { MongoClient, Db } from "mongodb";
+
+export default class AnalyticsDataSource {
+  private client: MongoClient;
+  private isConnected: boolean = false;
+  private db: Db;
+  static readonly dbName = "sample_analytics";
+  static readonly collTxns = "transactions";
+  static readonly fieldTxns = "transactions";
+
+  constructor(connString: string) {
+    this.client = new MongoClient(connString);
+  }
+
+  public async initialize() {
+    if (!this.isConnected) {
+      await this.client.connect();
+      this.db = this.client.db(AnalyticsDataSource.dbName);
+      this.isConnected = true;
+      console.log(`Connected to MongoDB: ${AnalyticsDataSource.dbName}`);
+    }
+  }
+
+  public async close(): Promise<void> {
+    await this.client.close();
+    this.isConnected = false;
+    console.log("MongoDB connection closed");
```

Comment:
Console should output this log once MongoDB connection gets closed See the PR section `Validation`